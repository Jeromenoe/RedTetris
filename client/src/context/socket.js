import React from 'react';
import socketio from "socket.io-client";


const localHost = "http://localhost:3001"; 
const herokuHost = "https://redtetris42-heroku.herokuapp.com"
const socketURL = process.env.NODE_ENV === "production" ? herokuHost : localHost

export const socket = socketio.connect(socketURL);
export const SocketContext = React.createContext();