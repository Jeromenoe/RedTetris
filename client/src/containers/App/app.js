import React, { useContext, useEffect } from 'react'
import { useSelector } from 'react-redux';
import Tetris from '../../components/Tetris/index';
import { SocketContext } from '../../context/socket';
import { AppContainer } from './styles'
import { Login } from '../Login/login'
import { WaitingRoom } from '../WaitingRoom/waitingRoom'
import { getGameInfos } from '../../selectors/game'
import { initSockets } from '../../sockets';

const App = () => {
	// console.log('re-render App');
	// const [playing, toggle] = useAudio('https://vgmsite.com/soundtracks/tetris-gameboy-rip/mpkrawiu/tetris-gameboy-02.mp3');
	const socket = useContext(SocketContext);
	const game = useSelector(getGameInfos)
	useEffect(() => {
		initSockets(socket);
	}, [socket]);
	let currentComponent = null;
	if (!game.name)
		currentComponent = <Login />
	else if (!game.isStarted)
		currentComponent = <WaitingRoom />
	else
		currentComponent = <Tetris />
	return (
		<AppContainer>
			{currentComponent}
			{/* <button onClick={toggle}>{playing ? "Pause" : "Play"}</button> */}
		</AppContainer>
	)
}

// const useAudio = url => {
//   const [audio] = useState(new Audio(url));
//   const [playing, setPlaying] = useState(false);

//   const toggle = () => setPlaying(!playing);

//   useEffect(() => {
//       playing ? audio.play() : audio.pause();
//     },
//     [playing]
//   );

//   useEffect(() => {
//     audio.addEventListener('ended', () => setPlaying(false));
//     return () => {
//       audio.removeEventListener('ended', () => setPlaying(false));
//     };
//   }, []);

//   return [playing, toggle];
// };

export default App;