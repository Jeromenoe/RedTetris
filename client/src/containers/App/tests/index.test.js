import React from 'react';
import App from '../app';

import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';
import toJson from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';

import { SocketContext, socket } from "../../../context/socket"


configure({adapter: new Adapter()});

describe('Testing App Container', () => {
	test('Check Login', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				name: null,
				isStarted: false,
				isOver: false,
				opponents: [],
				players: [],
				removeOwnIndestructibleLine: false,
				difficulty: 2,
				nbOfNextTetromino: 3,
			},
			me: {
				name: null,
				hasLost: false,
				stage: [[]],
				nextPieces: [],
				rows: 0,
				level: 0,
				score: 0
			}
		};
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><App /></SocketContext.Provider></Provider>);
		expect(toJson(wrapper)).toMatchSnapshot();
	  });

	  test('Check waiting room', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				name: 'test',
				isStarted: false,
				isOver: false,
				opponents: [],
				players: [
					{name: 'toto'}
				],
				removeOwnIndestructibleLine: false,
				difficulty: 2,
				nbOfNextTetromino: 3,
			},
			me: {
				name: 'toto',
				hasLost: false,
				stage: [[]],
				nextPieces: [],
				rows: 0,
				level: 0,
				score: 0
			}
		};
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><App /></SocketContext.Provider></Provider>);
		expect(toJson(wrapper)).toMatchSnapshot();
	  });

	  test('Check in game', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				name: 'test',
				isStarted: true,
				isOver: false,
				opponents: [],
				players: [
					[{name: 'toto'}]
				],
				removeOwnIndestructibleLine: false,
				difficulty: 2,
				nbOfNextTetromino: 3,
			},
			me: {
				name: 'toto',
				hasLost: false,
				stage: [[]],
				nextPieces: [],
				rows: 0,
				level: 0,
				score: 0
			}
		};
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><App /></SocketContext.Provider></Provider>);
		expect(toJson(wrapper)).toMatchSnapshot();
	  });
});