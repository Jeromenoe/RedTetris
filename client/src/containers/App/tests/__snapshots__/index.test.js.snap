// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Container Check Login 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App>
    <styled.div>
      <div
        className="sc-ksluID eUymye"
      >
        <Login>
          <styled.div>
            <div
              className="sc-fnVZcZ iijFWM"
            >
              <h1
                className="redTetrisTitle"
              >
                Red Tetris
              </h1>
              <styled.div>
                <div
                  className="sc-hBMUJo dHhYea"
                >
                  <input
                    className="userInfosTextField"
                    id="playerNameId"
                    placeholder="User Name"
                    type="text"
                  />
                </div>
              </styled.div>
              <styled.div>
                <div
                  className="sc-hBMUJo dHhYea"
                >
                  <input
                    className="userInfosTextField"
                    id="gameNameId"
                    placeholder="Game Name"
                    type="text"
                  />
                </div>
              </styled.div>
              <button
                className="PlayButton"
                onClick={[Function]}
              >
                PLAY
              </button>
            </div>
          </styled.div>
        </Login>
      </div>
    </styled.div>
  </App>
</Provider>
`;

exports[`Testing App Container Check in game 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App>
    <styled.div>
      <div
        className="sc-ksluID eUymye"
      >
        <Tetris>
          <styled.div
            id="styledTetrisWrapper"
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="sc-dlnjwi iPWQhR"
              id="styledTetrisWrapper"
              onKeyDown={[Function]}
              role="button"
              tabIndex="0"
            >
              <h1
                className="title"
                style={
                  Object {
                    "marginBottom": "40px",
                    "marginTop": "60px",
                  }
                }
              >
                RED TETRIS
              </h1>
              <styled.div>
                <div
                  className="sc-hKFxyN idlCam"
                >
                  <div
                    style={
                      Object {
                        "color": "rgb(55, 55, 55)",
                        "fontFamily": "'Press Start 2P', cursive",
                        "fontSize": "12px",
                        "margin": "5px 0",
                      }
                    }
                  >
                    toto
                    @
                    test
                  </div>
                  <div
                    style={
                      Object {
                        "background": "rgba(0,0,0,0.7)",
                        "height": "1px",
                        "width": "100%",
                      }
                    }
                  />
                  <styled.div>
                    <div
                      className="sc-jSFjdj djwQrH"
                    >
                      <div
                        style={
                          Object {
                            "display": "flex",
                          }
                        }
                      >
                        <PlayerData>
                          <div
                            style={
                              Object {
                                "alignSelf": "flex-end",
                                "marginBottom": "6px",
                                "marginRight": "6px",
                                "width": "70px",
                              }
                            }
                          >
                            <styled.div>
                              <div
                                className="sc-jrsJWt ilCZvT"
                              >
                                <div
                                  style={
                                    Object {
                                      "marginBottom": "15px",
                                    }
                                  }
                                >
                                  <div
                                    style={
                                      Object {
                                        "marginBottom": "6px",
                                      }
                                    }
                                  >
                                    Level
                                  </div>
                                  <span
                                    style={
                                      Object {
                                        "paddingRight": "6px",
                                      }
                                    }
                                  >
                                    0
                                  </span>
                                </div>
                                <div
                                  style={
                                    Object {
                                      "marginBottom": "15px",
                                    }
                                  }
                                >
                                  <div
                                    style={
                                      Object {
                                        "marginBottom": "6px",
                                      }
                                    }
                                  >
                                    Score
                                  </div>
                                  <span
                                    style={
                                      Object {
                                        "paddingRight": "6px",
                                      }
                                    }
                                  >
                                    0
                                  </span>
                                </div>
                                <div
                                  style={
                                    Object {
                                      "marginBottom": "20px",
                                    }
                                  }
                                >
                                  <div
                                    style={
                                      Object {
                                        "marginBottom": "6px",
                                      }
                                    }
                                  >
                                    Lines
                                  </div>
                                  <span
                                    style={
                                      Object {
                                        "paddingRight": "6px",
                                      }
                                    }
                                  >
                                    0
                                  </span>
                                </div>
                              </div>
                            </styled.div>
                          </div>
                        </PlayerData>
                        <aside>
                          <styled.div>
                            <div
                              className="sc-eCApnc kQHbck"
                            >
                              <Stage>
                                <styled.div
                                  cellSize={21}
                                  height={1}
                                  width={0}
                                >
                                  <div
                                    className="sc-gtsrHT cSPJwE"
                                    height={1}
                                    width={0}
                                  />
                                </styled.div>
                              </Stage>
                              <aside>
                                <NextPieces>
                                  <div>
                                    <styled.div>
                                      <div
                                        className="sc-gKAaRy jMDiMd"
                                      >
                                        <styled.h1>
                                          <h1
                                            className="sc-pNWdM chyiGP"
                                          >
                                            NEXT
                                          </h1>
                                        </styled.h1>
                                      </div>
                                    </styled.div>
                                  </div>
                                </NextPieces>
                              </aside>
                            </div>
                          </styled.div>
                        </aside>
                      </div>
                      <Opponents />
                      <button
                        className="LeaveButton"
                        id="leavePlayingGame"
                        onClick={[Function]}
                      >
                        LEAVE
                      </button>
                    </div>
                  </styled.div>
                </div>
              </styled.div>
              <Connect(SubLeaderbord)>
                <SubLeaderbord
                  dispatch={[Function]}
                  gameOver={false}
                >
                  <styled.div
                    gameOver={false}
                  >
                    <div
                      className="sc-dIsUp uSCOS"
                    >
                      <styled.div>
                        <div
                          className="sc-bqGGPW eLAWwL"
                        >
                          <strong
                            style={
                              Object {
                                "alignSelf": "center",
                                "borderBottom": "1px solid #ddd",
                                "display": "block",
                                "padding": "5px",
                                "width": "90%",
                              }
                            }
                          >
                            Game Over
                          </strong>
                          <div
                            className="player-score-container"
                          >
                            <div
                              className="player-score-rank"
                            >
                              Rank
                            </div>
                            <div
                              className="player-score-name"
                            >
                              Name
                            </div>
                            <div
                              className="player-score-pts"
                            >
                              Score
                            </div>
                            <div
                              className="player-score-lines"
                            >
                              Lines
                            </div>
                            <div
                              className="player-score-lvl"
                            >
                              Level
                            </div>
                          </div>
                          <div
                            className="player-score-container"
                            key="0"
                            style={
                              Object {
                                "fontWeight": "normal",
                              }
                            }
                          >
                            <div
                              className="player-score-rank"
                            />
                            <div
                              className="player-score-name"
                            />
                            <div
                              className="player-score-pts"
                            />
                            <div
                              className="player-score-lines"
                            />
                            <div
                              className="player-score-lvl"
                            />
                          </div>
                          <div
                            className="btn-leaderboard-container"
                          >
                            <button
                              className="btn-leave-leaderboard"
                              onClick={[Function]}
                            >
                              Waiting Room
                            </button>
                            <button
                              className="btn-leave-leaderboard"
                              id="btn-leave-in-game"
                              onClick={[Function]}
                            >
                              Leave
                            </button>
                          </div>
                        </div>
                      </styled.div>
                    </div>
                  </styled.div>
                </SubLeaderbord>
              </Connect(SubLeaderbord)>
            </div>
          </styled.div>
        </Tetris>
      </div>
    </styled.div>
  </App>
</Provider>
`;

exports[`Testing App Container Check waiting room 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App>
    <styled.div>
      <div
        className="sc-ksluID eUymye"
      >
        <WaitingRoom>
          <styled.div>
            <div
              className="sc-fFSPTT iltGLs"
            >
              <styled.div>
                <div
                  className="sc-iemWCZ iOZdrW"
                >
                  <h2
                    className="waitingRoomTitle"
                  >
                    TEST
                  </h2>
                  <button
                    className="settingsBtn"
                    id="showSettingPopup"
                  >
                    <img
                      alt="settings"
                      className="settingsImg"
                      onClick={[Function]}
                      src="./settingBtn.png"
                    />
                  </button>
                </div>
              </styled.div>
              <ul
                className="waitingPlayerList"
              >
                <styled.li
                  color="#f53643"
                  key="0"
                >
                  <li
                    className="sc-dIvrsQ cHitDp"
                    color="#f53643"
                  >
                    toto (me)
                  </li>
                </styled.li>
              </ul>
              <styled.div>
                <div
                  className="sc-bkbkJK gBqTsE"
                >
                  <button
                    className="LeaveButton"
                    onClick={[Function]}
                  >
                    LEAVE
                  </button>
                  <button
                    className="PlayButton"
                    onClick={[Function]}
                  >
                    PLAY
                  </button>
                </div>
              </styled.div>
              <GameSettings>
                <styled.div
                  id="gameSettingsPopup"
                >
                  <div
                    className="sc-hHEiqL eAUpup"
                    id="gameSettingsPopup"
                  >
                    <styled.div>
                      <div
                        className="sc-dlMDgC gYcacr"
                      >
                        <div
                          className="settings-header"
                        >
                          <span
                            className="CloseButton"
                            id="closebtn"
                            onClick={[Function]}
                          >
                            ×
                          </span>
                          <h2>
                            SETTINGS
                          </h2>
                        </div>
                        <ul
                          className="options-list"
                        >
                          <li
                            className="options-elem"
                            key="1"
                          >
                            DIFFICULTY:
                            <RadioOptions
                              settingId={1}
                            >
                              <div
                                className="radio-group"
                              >
                                <input
                                  checked={false}
                                  id="opt-1"
                                  name="selector-1"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-1"
                                >
                                  EASY
                                </label>
                                <input
                                  checked={true}
                                  id="opt-2"
                                  name="selector-1"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-2"
                                >
                                  MEDIUM
                                </label>
                                <input
                                  checked={false}
                                  id="opt-3"
                                  name="selector-1"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-3"
                                >
                                  HARD
                                </label>
                                <input
                                  checked={false}
                                  id="opt-4"
                                  name="selector-1"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-4"
                                >
                                  LEVEL
                                </label>
                              </div>
                            </RadioOptions>
                          </li>
                          <li
                            className="options-elem"
                            key="2"
                          >
                            NUMBER OF NEXT TETROMINOS:
                            <RadioOptions
                              settingId={2}
                            >
                              <div
                                className="radio-group"
                              >
                                <input
                                  checked={false}
                                  id="opt-5"
                                  name="selector-2"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-5"
                                >
                                  1
                                </label>
                                <input
                                  checked={false}
                                  id="opt-6"
                                  name="selector-2"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-6"
                                >
                                  2
                                </label>
                                <input
                                  checked={true}
                                  id="opt-7"
                                  name="selector-2"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-7"
                                >
                                  3
                                </label>
                                <input
                                  checked={false}
                                  id="opt-8"
                                  name="selector-2"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-8"
                                >
                                  4
                                </label>
                                <input
                                  checked={false}
                                  id="opt-9"
                                  name="selector-2"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-9"
                                >
                                  5
                                </label>
                              </div>
                            </RadioOptions>
                          </li>
                          <li
                            className="options-elem"
                            key="3"
                          >
                            DELETE OWN INDESTRUTIBLE LINES:
                            <RadioOptions
                              settingId={3}
                            >
                              <div
                                className="radio-group"
                              >
                                <input
                                  checked={true}
                                  id="opt-10"
                                  name="selector-3"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-10"
                                >
                                  FALSE
                                </label>
                                <input
                                  checked={false}
                                  id="opt-11"
                                  name="selector-3"
                                  onChange={[Function]}
                                  type="radio"
                                />
                                <label
                                  htmlFor="opt-11"
                                >
                                  TRUE
                                </label>
                              </div>
                            </RadioOptions>
                          </li>
                        </ul>
                      </div>
                    </styled.div>
                  </div>
                </styled.div>
              </GameSettings>
            </div>
          </styled.div>
        </WaitingRoom>
      </div>
    </styled.div>
  </App>
</Provider>
`;
